### ISSUE TITLE
[Question]: Less context than similarity_top_k

### ISSUE THREAD

**Issue by farrinfedra** [run-llama/llama_index#15396]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

 I'm encountering an issue where the context retrieved is less than the similarity_top_k value that I’ve set in my code. Additionally, I’ve noticed that when I run my code twice using the same index name (rebuilding the index from scratch each time), the amount of context retrieved is significantly reduced compared to when I run the program only once.  I'm not sure what's causing this discrepancy, and I would greatly appreciate any insights or assistance from @dosu or anyone else who might have experienced something similar.

**Comment by logan-markewich** [comment]
Sounds like you indexed duplicates of the same dara/chunks?

The retriever will remove duplicate texts

**Comment by farrinfedra** [comment]
> Sounds like you indexed duplicates of the same dara/chunks?
>  Even when I create a completely new index, I'm still retrieving fewer context chunks than expected. I believe someone else mentioned encountering a similar issue. According to the cosine similarity code, I should be receiving exactly similarity_top_k chunks, but that’s not happening. Could there be a threshold or some other mechanism in the code that’s cutting off or filtering out some of the chunks?
> The retriever will remove duplicate texts
I have seen the part of code where the duplicates are being removed, what I'm referring to is before where the duplicates are removed.

**Comment by logan-markewich** [comment]
Could you provide something to replicate the issue? A Google colab maybe?
